modules/order/src/Form/OrderForm.php

```diff
@@ -121,8 +121,13 @@ public function form(array $form, FormStateInterface $form_state) {
       $form['uid']['#group'] = 'customer';
     }
     else {
-      $user_link = $order->getCustomer()->toLink()->toString();
-      $form['customer']['uid'] = $this->fieldAsReadOnly($this->t('Customer'), $user_link);
+      if ($user = $order->getCustomer()) {
```

Comment:
This kind of fix is wrong way to fix this bug.
Propper way described in https://www.drupal.org/node/2898218#comment-12231013